src/modules/rust.rs

```diff
@@ -100,10 +114,11 @@ fn env_rustup_toolchain(context: &Context) -> Option<String> {
     Some(val.trim().to_owned())
 }
 
-fn execute_rustup_override_list(cwd: &Path) -> Option<String> {
+async fn execute_rustup_override_list(cwd: &Path) -> Option<String> {
     let Output { stdout, .. } = Command::new("rustup")
```

Comment:
I think `exec_cmd` already handles errors sufficiently by logging them. If you think the real error is required here, I'd prefer you add a second method to `Context`.
Beyond that: I think for modules that are async it makes sense to ignore `command_timeout` because having multiple separate timeouts for a single module would get confusing.